default.nix

```diff
@@ -202,34 +206,44 @@ let
         inherit pyProject;
         pythonPackages = py.pkgs;
       };
-    in
-    py.pkgs.buildPythonApplication
-      (
-        passedAttrs // {
-          pname = moduleName pyProject.tool.poetry.name;
-          version = pyProject.tool.poetry.version;
+      app = py.pkgs.buildPythonPackage
+        (
+          passedAttrs // {
+            pname = moduleName pyProject.tool.poetry.name;
+            version = pyProject.tool.poetry.version;
 
-          inherit src;
+            inherit src;
 
-          format = "pyproject";
+            format = "pyproject";
+            # Like buildPythonApplication, but without the toPythonModule part
+            # Meaning this ends up looking like an application but it also
+            # provides python modules
+            namePrefix = "";
 
-          buildInputs = mkInput "buildInputs" buildSystemPkgs;
-          propagatedBuildInputs = mkInput "propagatedBuildInputs" (getDeps "dependencies") ++ ([ py.pkgs.setuptools ]);
-          nativeBuildInputs = mkInput "nativeBuildInputs" [ pkgs.yj py.pkgs.removePathDependenciesHook ];
-          checkInputs = mkInput "checkInputs" (getDeps "dev-dependencies");
+            buildInputs = mkInput "buildInputs" buildSystemPkgs;
+            propagatedBuildInputs = mkInput "propagatedBuildInputs" (getDeps "dependencies") ++ ([ py.pkgs.setuptools ]);
+            nativeBuildInputs = mkInput "nativeBuildInputs" [ pkgs.yj py.pkgs.removePathDependenciesHook ];
+            checkInputs = mkInput "checkInputs" (getDeps "dev-dependencies");
 
-          passthru = {
-            python = py;
-          };
+            passthru = {
+              python = py;
+            };
 
-          meta = meta // {
-            inherit (pyProject.tool.poetry) description homepage;
-            inherit (py.meta) platforms;
-            license = getLicenseBySpdxId (pyProject.tool.poetry.license or "unknown");
-          };
+            meta = meta // {
+              inherit (pyProject.tool.poetry) description homepage;
+              inherit (py.meta) platforms;
+              license = getLicenseBySpdxId (pyProject.tool.poetry.license or "unknown");
+            };
 
-        }
-      );
+          }
+        );
+      appEnv = py.buildEnv.override {
+        extraLibs = poetryPython.poetryPackages ++ [ app ];
```

Comment:
This is a bit broad. In the case of `mkPoetryApplication` we don't want to include `dev-dependencies` in this call to `buildEnv`.

Reply:
Agreed, now fixed by having `mkPoetryPackages` expose a separate attribute for runtime-specific dependencies

Reply:
Oh duh, there's `getDeps "dependencies"` I can use instead. Doing that now